modules/client/src/internal/client/decoding.ts

```diff
@@ -133,18 +133,9 @@ export class ClientDecoding extends ClientCore implements IClientDecoding {
     }
     const result = daoInterface.decodeFunctionData("setMetadata", data);
     const bytes = hexToBytes(result[0]);
-    const contentUriText = new TextDecoder().decode(bytes);
-    const contentUri = new URL(contentUriText);
-    if (contentUri.protocol !== "ipfs:") {
-      throw new UnsupportedProtocolError(contentUri.protocol);
-    }
-    const cid = contentUri.host;
-    const ipfsRegex =
-      /^Qm([1-9A-HJ-NP-Za-km-z]{44,}|b[A-Za-z2-7]{58,}|B[A-Z2-7]{58,}|z[1-9A-HJ-NP-Za-km-z]{48,}|F[0-9A-F]{50,})$/;
-    if (!ipfsRegex.test(cid)) {
-      throw new Error("The metadata URL defined on the DAO is invalid");
-    }
-    return contentUriText;
+    const metadataUri = new TextDecoder().decode(bytes);
+    resolveIpfsCid(metadataUri)
```

Comment:
What's the use of calling this function without using any return value of it? Throwing if incorrect?